__tests__/jestGlobalSetup.ts

```diff
@@ -0,0 +1,255 @@
+/* eslint-disable no-console */
+/**
+ * Asynchronous Global Test Setup
+ * Run only once
+ * ref: order of execution jestGlobalSetup.ts -> jest.setup.ts -> fixtures.ts
+ */
+
+import { BaseUrl } from '../src/constants';
+
+type DevnetStrategy = {
+  isDevnet: boolean;
+  isRS: boolean;
+};
+type ProviderType = {
+  sequencer: boolean;
+  rpc: boolean;
+};
+
+/**
+ * Global Setup Fixtures
+ */
+
+/* Default test config based on run `starknet-devnet --seed 0` */
+const GS_DEFAULT_TEST_PROVIDER_URL = 'http://127.0.0.1:5050/';
+
+const overrideOrSet = (envName: string, setValue: string) => {
+  process.env[envName] = process.env[envName] ?? setValue;
+};
+
+const localDevnetDetectionStrategy = async () => {
+  const setup = (strategy: DevnetStrategy) => {
+    overrideOrSet('IS_LOCALHOST_DEVNET', strategy.isDevnet ? 'true' : 'false');
+    overrideOrSet(
+      'IS_RPC_DEVNET',
+      strategy.isDevnet && (strategy.isRS || process.env.TEST_RPC_URL) ? 'true' : 'false'
+    );
+    overrideOrSet(
+      'IS_SEQUENCER_DEVNET',
+      strategy.isDevnet && process.env.IS_RPC_DEVNET === 'false' ? 'true' : 'false'
+    );
+    return strategy;
+  };
+
+  const strategy: DevnetStrategy = {
+    isDevnet: false,
+    isRS: false,
+  };
+  // if is_alive work it is local devnet
+  try {
+    const response = await fetch(`${GS_DEFAULT_TEST_PROVIDER_URL}is_alive`);
+    const body = await response.text();
+    if (body === 'Alive!!!') {
+      strategy.isDevnet = true;
+    } else {
+      return setup(strategy);
+    }
+  } catch (error) {
+    return setup(strategy);
+  }
```

Comment:
An approach I sometimes use to condense simple `try/catch` blocks is to use a promise chain, it might be worth considering here. Same thing could also be done for the RPC request.

```suggestion
  const devnetResult = await fetch(`${GS_DEFAULT_TEST_PROVIDER_URL}is_alive`)
    .then((response) => response.text())
    .catch(() => '');
  if (devnetResult !== 'Alive!!!') {
    return setup(strategy);
  }
  strategy.isDevnet = true;
```

Reply:
Nice one